### PR TITLE
WIP add Request::initialize_as_target

### PR DESCRIPTION
**Pull request by dimpolo**
I took a stab at adding a `initialize_as_target` `Request` builder.

I'm afraid I currently don't know enough about this kind of functionality so I'll leave some questions here.
@averyanalex maybe you could shed some light on them?

### CODE REVIEW

File: src/requests.rs [hunk 1]
```diff
@@ -249,3 +299,64 @@ pub enum MifareCommand {
...
+
+/// The information needed to be able to be activated at 106 kbps in passive mode
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+pub struct TargetMifareParams {
+    /// 2 bytes LSB first, as defined in ISO/IEC14443-3
+    pub sens_res: [u8; 2],
+    /// has a fixed length of 3 bytes containing the nfcid11 to nfcid13 bytes.
+    /// Indeed, the PN532 can handle only nfcid1t in single size
+    pub nfcid1t: [u8; 3],
+    pub sel_res: TargetSelRes,
```
Comment (dimpolo) [inline src/requests.rs:339]: Are there some default values that make sense here?

File: src/requests.rs [hunk 2]
```diff
@@ -249,3 +299,64 @@ pub enum MifareCommand {
...
+
+/// The information to be able to respond to a polling request
+/// at 212/424 kbps in passive mode
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+pub struct TargetFelicaParams {
+    pub fncid2t: [u8; 8],
+    pub pad: [u8; 8],
+    /// These two bytes are returned in the POL_RES frame
+    /// if the 4th byte of the incoming POL_REQ command frame is 0x01.
+    pub system_code: [u8; 2],
```
Comment (dimpolo) [inline src/requests.rs:361]: Are there some default values that make sense here?

File: src/requests.rs [hunk 3]
```diff
@@ -249,3 +299,64 @@ pub enum MifareCommand {
...
     Transfer = 0xB0,
 }
+
+/// Indicates which target mode the PN532 should respect
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+#[repr(u8)]
+pub enum TargetMode {
+    // TODO is this a valid mode?
+    /// Accept all target modes
+    Any = 0b000,
```
[the remark this answers is not in the document]
Comment (averyanalex) [reply src/requests.rs:309]: I think yes: there is no SyntaxError, and chip responds when I bring android phone to pn532 when target_mode is TargetMode::Any

File: src/requests.rs [hunk 4]
```diff
@@ -93,6 +93,56 @@ impl Request<0> {
...
+                nfcid3t[2],
+                nfcid3t[3],
+                nfcid3t[4],
+                nfcid3t[5],
+                nfcid3t[6],
+                nfcid3t[7],
+                nfcid3t[8],
+                nfcid3t[9],
+                0, // no general bytes
+                0, // no historical bytes
```
[the remark this answers is not in the document]
Comment (averyanalex) [reply src/requests.rs:142]: If lib will just skip this in TgInitAsTarget command, there will be no way to set these later, becouse lib will wait chip's response, so you can't send TgSetGeneralBytes command just after TgInitAsTarget

**Comment by averyanalex** [comment]
What I can say about default values: with everything set to [0; N], TargetMode::PassiveOnly and TargetSelRes::PICC, my Android phone detects FeliCa: 
IMG_20230130_230001

When I set params to [some random recommendations from stackoverflow](https://stackoverflow.com/a/59714334/20666749), NFC Tools can't see pn532, but it responses to microcontroller. (so there is a connection, but chip is not recognized by phone).

Also I don't know how to emulate ISO 14443-3A (but I want to do it).

**Comment by dimpolo** [comment]
Just thought I'd let you know. I'm rather limited on time currently so it might take me some weeks to properly finish this.

(btw, I spent like 4 hours today on https://github.com/esp-rs/esp-hal/issues/295#issuecomment-1410885751 which you also seem to have encountered 🤣)

**Comment by dimpolo** [comment]
Being honest, I don't think I'll find the time to finish this.
If anyone's interested they can continue from here :)

**Pull request closed without merging**